src/MicroPlate.php

```diff
@@ -27,18 +34,133 @@ class MicroPlate
      */
     public function __construct(CoordinateSystem $coordinateSystem)
     {
-        $this->wells = new Collection([]);
         $this->coordinateSystem = $coordinateSystem;
+
+        $this->clearWells();
     }
 
     /**
-     * TODO how to deal with duplicates?
-     *
      * @param Coordinate<TCoordinateSystem> $coordinate
      * @param TWell $content
      */
     public function addWell(Coordinate $coordinate, $content): void
     {
-        $this->wells->add([$coordinate, $content]);
+        $this->assertWellIsEmpty($coordinate, $content);
+        $this->forceAddWell($coordinate, $content);
+    }
+
+    /**
+     * @param TWell $content
+     */
+    public function addToNextFreeWell($content, FlowDirection $flowDirection): void
+    {
+        $coordinate = $this->nextFreeWellCoordinate($flowDirection);
+        $this->addWell($coordinate, $content);
+    }
+
+    public function clearWells(): void
+    {
+        $this->wells = new Collection([]);
+        foreach ($this->coordinateSystem->columns() as $column) {
```

Comment:
In my opinion, there are advantages and disadvantages:
a) If we consider the flow direction, then we would have to include it in the constructor of the microplate. Thus, a microplate would have a flow direction. For me, however, this is not a property of the plate itself but a parameter of the access to the plate.

b) If we ignore it we would have a kind of natural/implicit sorting and wouldn't have to give the flow direction with every e.g. position-access.

Personally, I would argue for a), so that the user always specifies the flow direction when accessing the plate.

Reply:
Makes sense, would be good to have it documented in a code comment.